front/components/agent_builder/observability/charts/LatencyChart.tsx

```diff
@@ -0,0 +1,159 @@
+import { useMemo } from "react";
+import {
+  Area,
+  AreaChart,
+  CartesianGrid,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from "recharts";
+import type { TooltipContentProps } from "recharts/types/component/Tooltip";
+
+import {
+  CHART_HEIGHT,
+  LATENCY_LEGEND,
+  LATENCY_PALETTE,
+} from "@app/components/agent_builder/observability/constants";
+import { useObservability } from "@app/components/agent_builder/observability/ObservabilityContext";
+import { ChartContainer } from "@app/components/agent_builder/observability/shared/ChartContainer";
+import { ChartLegend } from "@app/components/agent_builder/observability/shared/ChartLegend";
+import { ChartTooltipCard } from "@app/components/agent_builder/observability/shared/ChartTooltip";
+import { useAgentLatency } from "@app/lib/swr/assistants";
+
+interface LatencyData {
+  messages: number;
+  average: number;
+}
+
+function isLatencyData(data: unknown): data is LatencyData {
+  return typeof data === "object" && data !== null && "average" in data;
+}
+
+function LatencyTooltip(
```

Comment:
IIRC I moved all tooltips inside "/charts/ChartsTooltip.tsx", wdyt moving this one there?